packages/starknet-snap/src/rpcs/executeTxn.test.ts

```diff
@@ -0,0 +1,190 @@
+import type { Invocations, UniversalDetails } from 'starknet';
+import { TransactionType, constants } from 'starknet';
+
+import invocationExamples from '../__tests__/fixture/invocationExamples.json'; // Assuming you have a similar fixture
+import { STARKNET_SEPOLIA_TESTNET_NETWORK } from '../utils/constants';
+import * as starknetUtils from '../utils/starknetUtils';
+import { executeTxn as executeTxnUtil } from '../utils/starknetUtils';
+import {
+  mockAccount,
+  prepareConfirmDialog,
+  prepareMockAccount,
+} from './__tests__/helper';
+import type { ExecuteTxnParams } from './executeTxn';
+import { executeTxn } from './executeTxn';
+
+const prepareMockExecuteTxn = (
+  transactionHash: string,
+  accountDeployed: boolean,
+) => {
+  const executeTxnRespMock = {
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    transaction_hash: transactionHash,
+  };
+
+  const getEstmatedFeesRepsMock = {
+    suggestedMaxFee: BigInt(1000000000000000).toString(10),
+    overallFee: BigInt(1000000000000000).toString(10),
+    includeDeploy: !accountDeployed,
+    unit: 'wei',
+  };
+  const getEstimatedFeesStub = jest.spyOn(starknetUtils, 'getEstimatedFees');
+  getEstimatedFeesStub.mockResolvedValue(getEstmatedFeesRepsMock);
+
+  const executeTxnUtilStub = jest.spyOn(starknetUtils, 'executeTxn');
+  executeTxnUtilStub.mockResolvedValue(executeTxnRespMock);
+
+  const createAccountStub = jest.spyOn(starknetUtils, 'createAccount');
+  createAccountStub.mockResolvedValue({
+    // eslint-disable-next-line @typescript-eslint/naming-convention
+    transaction_hash: '0x123',
+    address: '0x234',
+  });
+
+  return {
+    createAccountStub,
+    executeTxnRespMock,
+    getEstimatedFeesStub,
+    getEstmatedFeesRepsMock,
+  };
+};
+
+describe('ExecuteTxn', () => {
+  let invocationExample: any;
+  const enableAuthorize = true;
+  const state = {
+    accContracts: [],
+    erc20Tokens: [],
+    networks: [STARKNET_SEPOLIA_TESTNET_NETWORK],
+    transactions: [],
+  };
+
+  const createRequestParam = (
+    chainId: constants.StarknetChainId,
+    address: string,
+    invocations: Invocations,
+    details: UniversalDetails,
+    transactionType?: TransactionType,
+  ): ExecuteTxnParams => {
+    if (transactionType) {
+      invocations.forEach((invocation) => {
+        invocation.type = transactionType;
+      });
+    }
+    const request: ExecuteTxnParams = {
+      chainId,
+      address,
+      invocations,
+      details,
+    } as ExecuteTxnParams;
+    if (enableAuthorize) {
+      request.enableAuthorize = enableAuthorize;
+    }
+    return request;
+  };
+
+  it('executes transaction correctly when account is deployed', async () => {
+    const account = await mockAccount(constants.StarknetChainId.SN_SEPOLIA);
+    invocationExample = invocationExamples.shift();
```

Comment:
in JS, array is pass by reference, if u unshift here, it will affect the rest of the use on invocationExamples

if u need the first element, use invocationExamples[0]